packages/components/CHANGELOG.md

```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Breaking changes
+
+- Removed the deprecated position and menuLable from the DropdownMenu component.
+- Removed the deprecated onClickOutside prop from the Popover component.
```

Comment:
Bunch of nits:

- adding an empty line at the end of the block;
- Adding more empty spaces after each bullet point (as in the rest of the file)
- formatting prop names with backticks
- fixing a typo in `menuLabel`
- adding links to the PR introducing the change

```suggestion
### Breaking changes

-   Removed the deprecated `position` and `menuLabel` from the `DropdownMenu` component ([#34537](https://github.com/WordPress/gutenberg/pull/34537)).
-   Removed the deprecated `onClickOutside` prop from the `Popover` component ([#34537](https://github.com/WordPress/gutenberg/pull/34537)).

```